src/operators/ancilla.jl

```diff
@@ -0,0 +1,34 @@
+"""
+Attach the ancilla space `Va` and merge with physical space `Vp` of
+the physical operator `op`, so the new physical space becomes `Vp ⊗ Va`.
+"""
+function attach_ancilla(
+        op::AbstractTensorMap{T, S, N, N}; Va0::Union{ElementarySpace, Nothing} = nothing
+    ) where {T, S <: ElementarySpace, N}
+    Vp = first(codomain(op))
+    Va = isnothing(Va0) ? Vp' : Va0
+    id_Va = TensorKit.id(Va)
```

Comment:
I think you can create the identity for all spaces at the same time, to avoid the reduce